src/main/java/domain/Ladders.java

```diff
@@ -4,21 +4,34 @@
 import java.util.List;
 
 public class Ladders {
-    private final Height height;
     private final List<Ladder> ladders;
 
-    public Ladders(Height height) {
-        this.height = height;
+    public Ladders(int width, Height height, BooleanGenerator booleanGenerator) {
```

Comment:
```suggestion
        private Ladders(List<Ladder> ladders) {
        this.ladders = ladders;
    }

    public static Ladders of(int width, Height height, BooleanGenerator booleanGenerator) {
        return new Ladders(generate(width, height, booleanGenerator));
    }

    private static List<Ladder> generate(int width, Height height, BooleanGenerator booleanGenerator) {
        List<Ladder> ladders = new ArrayList<>();
        for (int i = 0; i < height.getHeight(); i++) {
            ladders.add(new Ladder(width, booleanGenerator));
        }
        return ladders;
    }
```

생성자에서 다른 도메인을 많이 받을수록 다른 도메인의 변경이 생성자에까지 영향을 미치게 될 것 같아요~ 매개변수가 많을 경우에는 [정적 팩터리 메서드](https://tecoble.techcourse.co.kr/post/2020-05-26-static-factory-method/)를 활용해보는 것도 방법일 것 같아요. 

온전한 객체를 생성자로 주입해주는 것도 방법이지만, 생성의 과정 중에 `generate` 내부의 기능들이 포함되어있다면 저는 지금처럼 생성자에서 generate 하는 것도 나쁘지 않다고 생각해요!!

Reply:
정적 팩토리 매서드를 쓰니깐 깔금해지고 영향도 덜 미쳐서 좋은것같아요!

```java
private final InputView inputView;
    private final OutputView outputView;
    private final Result result;

    private ResultController(InputView inputView, OutputView outputView, Result result) {
        this.inputView = inputView;
        this.outputView = outputView;
        this.result = result;
    }
    
    public static ResultController of(InputView inputView, OutputView outputView, Result result) {
        return new ResultController(inputView, outputView, result);
    }
```

ResultController 도 위와 같이 정적 팩토리를 사용하는게 좋을까요?

Reply:
ㅋㅋ 사용하면 안되는 경우는 없습니다!! 흠..  팩토리 메서드만 존재하는 클래스를 생성할 경우 상속이 불가하다는 단점 정도??